engine/kv_engine.hpp

```diff
@@ -157,20 +158,18 @@ class KVEngine : public Engine {
 
   inline Status MaybeInitAccessThread();
 
-  void SetCompareFunc(
-      const StringView& collection_name,
-      std::function<int(const StringView& src, const StringView& target)>
-          comp_func) {
-    comparator_.SetComparaFunc(collection_name, comp_func);
+  bool RegisterComparator(const StringView& collection_name,
+                          Comparator comp_func) {
+    return comparators_.RegisterComparator(collection_name, comp_func);
   }
 
-  Status CreateSortedCollection(const StringView collection_name,
-                                Collection** collection_ptr,
-                                const StringView& comp_name) override;
+  Status CreateSortedCollection(
+      const StringView collection_name, Collection** collection_ptr,
+      const SortedCollectionConfigs& configs) override;
 
  private:
   Status InitCollection(const StringView& collection, Collection** list,
-                        RecordType collection_type);
+                        RecordType collection_type, void* collection_configs);
```

Comment:
seems you removed InitCollection define. please removed here, or add its function define, or add TODO comments.